soccer/gameplay/robocup-py.cpp

```diff
@@ -715,8 +741,12 @@ BOOST_PYTHON_MODULE(robocup) {
         .def("min_y", &Geometry2d::Rect::miny)
         .def("max_x", &Geometry2d::Rect::maxx)
         .def("max_y", &Geometry2d::Rect::maxy)
+        .def("corners", &Geometry2d::Rect::corners)
+        .def("points", &Geometry2d::Rect::pointList)
+        .def("pad", &Geometry2d::Rect::pad)
         .def("near_point", &Geometry2d::Rect::nearPoint)
         .def("intersects_rect", &Geometry2d::Rect::intersects)
+        .def("intersects_segment", &Rect_segment_intersection)
```

Comment:
Just to make the naming convention the same between pts, lines, and segments, this should flip. IE: `segment_intersection` instead of `intersects_segment`. Feel free to change up  `intersects_rect` as well since you need to change the function it points to. It'll need a wrapper similar to how you do `segment` intersection

Reply:
I'm not sure why it needs a wrapper?  I just changed the name and it can call the same C++ function.

Reply:
Long story short, the callbacks boost does to convert a python function call to a C++ function call doesn't really deal with types super well. I believe it uses the type definition of the target function to figure out what to convert the python objects to. If there are two definitions with the same name, it doesn't know which one to pick and dies. If you use the wrapper functions, it picks the correct one

Reply:
got it- it's required because of the overloading only.  It wouldn't have been with the underscore